src/zmalloc.c

```diff
@@ -409,8 +423,11 @@ char *zstrdup(const char *s) {
 }
 
 size_t zmalloc_used_memory(void) {
-    size_t um;
-    atomicGet(used_memory,um);
+    assert(total_active_threads < MAX_THREADS_NUM);
+    size_t um = 0;
+    for (int i = 0; i < total_active_threads; i++) {
+        um += used_memory_thread[i];
```

Comment:
I think used_memory_thread[i] can return corrupted data if thread i is updating the value at the same time